### PR TITLE
Retrofit components

### PR DESCRIPTION
**Pull request by zhanwang626**
Ensure all tasks are retrofitted with their component type for rendering
issue: #1053

### CODE REVIEW

**Comment by scalvert** [comment]
This is a great start! One thing we need to do is ensure that all `component` definitions are under a parent `rule` property. That way when we call `addResult`, the `rule` is merged with the other rule metadata.

Example:

```ts
      this.addResult('some message',
        'review',
        'warning',
        {
          location: {
            uri: filePath,
            startColumn: column,
            startLine: line,
          },
          properties: {
            //...
          },
		  rule: {
		    properties: {
              component: 'table',
            }
          }
        }
      );
```

**Comment by zhanwang626** [comment]
> This is a great start! One thing we need to do is ensure that all `component` definitions are under a parent `rule` property. That way when we call `addResult`, the `rule` is merged with the other rule metadata.
> 
> Example:
> 
> ```ts
>       this.addResult('some message',
>         'review',
>         'warning',
>         {
>           location: {
>             uri: filePath,
>             startColumn: column,
>             startLine: line,
>           },
>           properties: {
>             // properties that are added to each result
>           },
> 		  rule: {
> 		    properties: { // properties that are added to the rule metadata
>               component: 'table',
>             }
>           }
>         }
>       );
> ```

Yeah, I just make sure all of the component definitions are under a parent rule property and updated this PR.

File: packages/checkup-plugin-ember/src/tasks/ember-dependencies-task.ts [hunk 1]
```diff
@@ -26,11 +26,14 @@ export default class EmberDependenciesTask extends BaseTask implements Task {
...
-              packageName: dependency.packageName,
-              packageVersion: dependency.packageVersion,
-              latestVersion: dependency.latestVersion,
-              type: dependency.type,
+            rule: {
+              properties: {
+                packageName: dependency.packageName,
+                packageVersion: dependency.packageVersion,
+                latestVersion: dependency.latestVersion,
+                type: dependency.type,
```
Comment (scalvert) [inline packages/checkup-plugin-ember/src/tasks/ember-dependencies-task.ts:34]: These properties shouldn't be under `rule.properties`. Only the `component` definition, but the others should be back up one level where they were before. They pertain to the result, not the rule.

File: packages/checkup-plugin-ember/src/tasks/ember-octane-migration-status-task.ts [hunk 2]
```diff
@@ -226,6 +226,7 @@ export default class EmberOctaneMigrationStatusTask extends BaseTask implements
             properties: {
               parentRuleID: this.taskName,
               taskDisplayName: `Ember Octane Migration | ${ruleMetadata.feature}`,
+              component: 'table',
```
Comment (scalvert) [inline packages/checkup-plugin-ember/src/tasks/ember-octane-migration-status-task.ts:229]: We should wait until #1060 is merged as this definition has changed.

File: packages/checkup-plugin-javascript/src/tasks/outdated-dependencies-task.ts [hunk 3]
```diff
@@ -27,11 +27,14 @@ export default class OutdatedDependenciesTask extends BaseTask implements Task {
...
-              packageName: dependency.packageName,
-              packageVersion: dependency.packageVersion,
-              latestVersion: dependency.latestVersion,
-              type: dependency.type,
+            rule: {
+              properties: {
+                packageName: dependency.packageName,
+                packageVersion: dependency.packageVersion,
+                latestVersion: dependency.latestVersion,
+                type: dependency.type,
```
Comment (scalvert) [inline packages/checkup-plugin-javascript/src/tasks/outdated-dependencies-task.ts:35]: Similar to the other comment above, these don't belong here. They're part of the top-level `properties`, not the `rule.properties`.

**Review by scalvert — CHANGES_REQUESTED** [review]
As an aside, I'm not sure we should be setting everything to use `table`. I wouldn't expect that'd be the primary output we'd want for all tasks. We should discuss this, but we should likely chose something a bit richer.

File: packages/checkup-plugin-ember/src/tasks/ember-dependencies-task.ts (hunk 1 above)
Comment (zhanwang626) [reply packages/checkup-plugin-ember/src/tasks/ember-dependencies-task.ts:34]: Got it.

**Comment by zhanwang626** [comment]
> As an aside, I'm not sure we should be setting everything to use `table`. I wouldn't expect that'd be the primary output we'd want for all tasks. We should discuss this, but we should likely chose something a bit richer.

Sure, we can set the default component to `task-result-list` which can display more details about nested infomation of a task result.

**Review by scalvert — APPROVED** [review]

**Pull request merged**